logr.go

```diff
@@ -430,3 +481,26 @@ type CallDepthLogSink interface {
 	// Successive calls to this are additive.
 	WithCallDepth(depth int) LogSink
 }
+
+// CallStackHelperLogSink represents a Logger that knows how to climb
+// the call stack to identify the original call site and can skip
+// intermediate helper functions if they mark themselves as
+// helper. Go's testing package uses that approach.
+//
+// This is useful for users who have helper functions between the
+// "real" call site and the actual calls to Logger methods.
+// Implementations that log information about the call site (such as
+// file, function, or line) would otherwise log information about the
+// intermediate helper functions.
+//
+// This is an optional interface and implementations are not required
+// to support it. Implementations that choose to support this must not
+// simply implement it as WithCallDepth(1), because
+// Logger.WithCallStackHelper will call both methods if they are
+// present. This should only be implemented for LogSinks that actually
+// need it, as with testing.T.
+type CallStackHelperLogSink interface {
+	// GetHelper returns a function that must be called to mark the direct caller
+	// as helper function when logging call site information.
+	GetHelper() func()
```

Comment:
should this be `GetHelperFunc()` or `GetCallStackHelper()` ?

Reply:
I wasn't sure. If we want a longer name, then I prefer `GetCallStackHelper` because it is more consistent.

Reply:
I've pushed a version with `GetCallStackHelper`.